pkg/webui/console/components/pubsub-form/index.js

```diff
@@ -69,23 +74,33 @@ export default class PubsubForm extends Component {
 
     const initialIsMqtt = update && 'mqtt' in initialPubsubValue
     const initialMqttSecure = initialIsMqtt ? initialPubsubValue.mqtt.use_tls : false
+    const initialUseCredentialsMqtt = initialIsMqtt
+      ? Boolean(initialPubsubValue.mqtt.username && initialPubsubValue.mqtt.password)
+      : true
+
+    const initialIsNats = update && 'nats' in initialPubsubValue
+    const { password, username, secure } = initialIsNats
+      ? mapNatsServerUrlToFormValue(initialPubsubValue.nats.server_url)
+      : { password: undefined, username: undefined, secure: false }
+    const initialNatsSecure = initialIsNats ? secure : false
+    const initialUseCredentialsNats = initialNatsSecure ? Boolean(password && username) : true
```

Comment:
see my review comment

```suggestion
    const initialUseCredentialsNats = initialNatsSecure ? Boolean(password || username) : true
```